### PR TITLE
#328 shift item code

### LINKED ISSUE #328: Shift item related code to `system/item`
- The item related queries and components could be shifted to the `system` package to make them more reusable by other packages

### PR DESCRIPTION
**Pull request by joshxg**
Fixes #328 

*Sigh* - Coin jar for each mess up with the generated file diff?

- I made `ItemSearchInput` in the `system/items` and all the queries are now in there
- I separated the `ItemListView` and general item queries, as we don't need to query for the stock lines for the list view

### CODE REVIEW

File: packages/system/src/Item/Components/ItemSearchInput/ItemSearchInput.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,24 @@
+import React from 'react';
+import { Story } from '@storybook/react';
+import { ItemSearchInput } from './ItemSearchInput';
+import { Item } from '@openmsupply-client/common';
+
+export default {
+  title: 'Item/ItemSearchInput',
```
Comment (mark-prins) [inline packages/system/src/Item/Components/ItemSearchInput/ItemSearchInput.stories.tsx:7]: might be better under `Inputs` - we will end up with a lot of stories I expect, and grouping them nicely is going to be a problem.

**Review by mark-prins — APPROVED** [review]
👍  That's much better - like that a lot!

**Comment by mark-prins** [comment]
Couple of afterthoughts
- the sorting on the item table wasn't working properly, in that it doesn't indicate what is currently sorted
- I expect we'll need pagination ( & filtering? ) since we'll be showing a big list of items under 'catalogue'

File: packages/system/src/Item/Components/ItemSearchInput/ItemSearchInput.stories.tsx (hunk 1 above)
Comment (joshxg) [reply packages/system/src/Item/Components/ItemSearchInput/ItemSearchInput.stories.tsx:7]: Heh I had the same thought and my version of grouping nicely was to put it under the type 🤣 

I do think the story book will likely get massive. I figured if someone was to look for a component they could use to search for items or names, looking under a names or items header. Though at the same time if someone was looking for a component they could copy to create an autocomplete for looking up periods, then they might look at inputs.

I'm not sure, but I can also see `inputs` become massive in itself and needing to be separated out. I'd like to keep it like this for now? It's just changing the string from `Item` to `Inputs` so should be an easy change when it's more obvious you're right? 😁

**Comment by joshxg** [comment]
> I expect we'll need pagination ( & filtering? ) since we'll be showing a big list of items under 'catalogue'

The catalogue list view already has pagination and sorting. Nothing has filtering yet, but yes I think we need to add a search bar to all of the list views.

I also am curious what's going to happen with the customers/item searchinputs/modals fetching all of the records i.e. the item input in the `ItemDetailsModal`. I suspect we'll need to use server side filtering there

**Pull request merged**